Simplenote/Classes/UITableView+Simplenote.swift

```diff
@@ -46,3 +46,120 @@ extension UITableView {
         return dequeueReusableHeaderFooterView(withIdentifier: T.reuseIdentifier) as? T
     }
 }
+
+// MARK: - Navigation
+//
+extension UITableView {
+
+    /// Returns first index path from the first non-empty section
+    ///
+    var firstIndexPath: IndexPath? {
+        let indexPath = IndexPath(row: 0, section: 0)
+        if numberOfRows(inSection: 0) == 0 {
+            return nextIndexPath(after: indexPath)
+        }
+
+        return indexPath
+    }
+
+    /// Returns index path after currently selected index path
+    ///
+    var nextIndexPath: IndexPath? {
+        guard let indexPath = indexPathForSelectedRow else {
+            return nil
+        }
+
+        return nextIndexPath(after: indexPath)
+    }
+
+    /// Returns index path before currently selected index path
+    ///
+    var prevIndexPath: IndexPath? {
+        guard let indexPath = indexPathForSelectedRow else {
+            return nil
+        }
+
+        var row = indexPath.row
+        var section = indexPath.section
+
+        while true {
+            if row == 0 {
+                if section == 0 {
+                    return nil
+                }
+
+                section -= 1
+                row = numberOfRows(inSection: section)
+            } else {
+                return IndexPath(row: row - 1, section: section)
+            }
+        }
+    }
+
+    /// Selects row after currently selected row or first row if no row is currently selected
+    ///
+    func selectNextRow() {
+        guard let indexPath = (indexPathForSelectedRow != nil ? nextIndexPath : firstIndexPath) else {
+            return
+        }
+        deselectSelectedRow()
+        selectRow(at: indexPath, animated: false, scrollPosition: .none)
+        scrollRectToVisible(rectForRow(at: indexPath), animated: false)
+    }
+
+    /// Selects row before currently selected row or first row if no row is currently selected
+    ///
+    func selectPrevRow() {
+        guard let indexPath = prevIndexPath ?? firstIndexPath else {
+            return
+        }
+        deselectSelectedRow()
+        selectRow(at: indexPath, animated: false, scrollPosition: .none)
+        scrollRectToVisible(rectForRow(at: indexPath), animated: false)
+    }
+
+    /// Calls `didSelectRow` for a currently selected row
+    ///
+    func executeSelection() {
+        guard let indexPath = indexPathForSelectedRow else {
+            return
+        }
+
+        delegate?.tableView?(self, didSelectRowAt: indexPath)
+    }
+
+    /// Deselects selected row if any
+    ///
+    func deselectSelectedRow() {
+        guard let indexPath = indexPathForSelectedRow else {
+            return
+        }
+
+        deselectRow(at: indexPath, animated: false)
+    }
+}
+
+// MARK: - Navigation (Private)
+//
+private extension UITableView {
+    func nextIndexPath(after prevIndexPath: IndexPath) -> IndexPath? {
```

Comment:
Nipticky: Hooow about:

```
    func indexPath(after indexPath: IndexPath) -> IndexPath? {
Write
```

Since the direction is implied by the `after` ?

Reply:
Good point, done!